json_serializable/test/src/json_serializable_test_input.dart

```diff
@@ -214,3 +214,13 @@ class SuperType {
   int priceFraction(int other) =>
       superTypeViaCtor == null ? null : superTypeViaCtor ~/ other;
 }
+
+@JsonSerializable(createFactory: false)
+class TrivialNestedNullable {
+  TrivialNestedNullable child;
```

Comment:
consider adding an `int` or some other non-custom type here, we want to make sure we don't call `toJson` on those

Reply:
sure...the code paths for this are so crazy independent, I'm not too worried – but paranoia is good!

Reply:
It also prevents future regressions etc, looking at this PR alone its not clear this has the correct behavior today.

Reply:
done